### ISSUE TITLE
Add support for compilation with gmake on FreeBSD

### ISSUE THREAD

**Issue by c0decafe** [lovell/sharp#326]
Building on freebsd fails with the below:

```shell
node-gyp rebuild
gmake: Entering directory '/home/downloads/sharp/build'
  CXX(target) Release/obj.target/sharp/src/common.o
../src/common.cc:4:10: fatal error: 'vips/vips.h' file not found
```

looking at the relevant bits of build/sharp.target.mk:

```makefile
INCS_Release := \
        -I/home/.node-gyp/4.1.1/include/node \
        -I/home/.node-gyp/4.1.1/src \
        -I/home/.node-gyp/4.1.1/deps/uv/include \
        -I/home/.node-gyp/4.1.1/deps/v8/include \
        -I$(srcdir)/node_modules/nan \
        -I$(srcdir)/-I/usr/local/include -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include/libxml2 -I/usr/local/include/libgsf-1 -I/usr/local/include/GraphicsMagick -I/usr/local/include/orc-0.4 -I/usr/local/include/OpenEXR -I/usr/local/include/pango-1.0 -pthread -I/usr/local/include/freetype2  -I/usr/local/include/harfbuzz -I/usr/local/include/libpng16
```

notice **-I$(srcdir)/-I**/usr/local/include

as i'm not familiar with gyp, i couldn't think of better than the below, inspired by https://github.com/Automattic/node-canvas/blob/master/binding.gyp:
```patch
diff --git a/binding.gyp b/binding.gyp
index 3538f8e..cd66b2f 100644
--- a/binding.gyp
+++ b/binding.gyp
@@ -49,7 +49,7 @@
     'conditions': [
       ['use_global_vips == "true"', {
         # Use pkg-config for include and lib
-        'include_dirs': ['<!(PKG_CONFIG_PATH="<(pkg_config_path)" pkg-config --cflags vips glib-2.0)'],
+        'include_dirs': ['<!(PKG_CONFIG_PATH="<(pkg_config_path)" pkg-config --cflags-only-I vips glib-2.0 | sed s/-I// )'],
         'conditions': [
           ['runtime_link == "static"', {
             'libraries': ['<!(PKG_CONFIG_PATH="<(pkg_config_path)" pkg-config --libs vips --static)']
```

```shell
% uname -svm
FreeBSD FreeBSD 10.2-RELEASE #0 r286666: Wed Aug 12 15:26:37 UTC 2015     root@releng1.nyi.freebsd.org:/usr/obj/usr/src/sys/GENERIC  amd64

% pkg-config --cflags vips glib-2.0
-I/usr/local/include -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include -I/usr/local/include/libxml2 -I/usr/local/include/libgsf-1 -I/usr/local/include/GraphicsMagick -I/usr/local/include/orc-0.4 -I/usr/local/include/OpenEXR -I/usr/local/include/pango-1.0 -pthread -I/usr/local/include/freetype2  -I/usr/local/include/harfbuzz -I/usr/local/include/libpng16
```

**Comment by lovell** [comment]
Hello, you are very likely to be the first person to try to use sharp on FreeBSD, so thank you for investigating!

The problem looks a bit like gyp is treating `include_dirs` as a scalar/string rather than an array of entries. I think gyp provides an `@` operator that might help here - let me take a closer look.

Are you aware of any continuous integration services that use/provide FreeBSD? I'd be much more comfortable saying "FreeBSD is supported" if we can automagically test against it.

**Comment by c0decafe** [comment]
the @ operator does help, but it still needs piping to sed:

```gyp
include_dirs': ['<!@(PKG_CONFIG_PATH="<(pkg_config_path)" pkg-config --cflags-only-I vips glib-2.0)'],
```
```makefile
INCS_Release := \
        -I/home/.node-gyp/4.1.1/include/node \
        -I/home/.node-gyp/4.1.1/src \
        -I/home/.node-gyp/4.1.1/deps/uv/include \
        -I/home/.node-gyp/4.1.1/deps/v8/include \
        -I$(srcdir)/node_modules/nan \
        -I$(srcdir)/-I/usr/local/include \
        -I$(srcdir)/-I/usr/local/include/glib-2.0 \
        -I$(srcdir)/-I/usr/local/lib/glib-2.0/include \
        -I$(srcdir)/-I/usr/local/include/libxml2 \
        -I$(srcdir)/-I/usr/local/include/libgsf-1 \
        -I$(srcdir)/-I/usr/local/include/GraphicsMagick \
        -I$(srcdir)/-I/usr/local/include/orc-0.4 \
        -I$(srcdir)/-I/usr/local/include/OpenEXR \
        -I$(srcdir)/-I/usr/local/include/pango-1.0 \
        -I$(srcdir)/-I/usr/local/include/freetype2 \
        -I$(srcdir)/-I/usr/local/include/harfbuzz \
        -I$(srcdir)/-I/usr/local/include/libpng16
```


```gyp
include_dirs': ['<!@(PKG_CONFIG_PATH="<(pkg_config_path)" pkg-config --cflags-only-I vips glib-2.0 | sed s\/-I//g )'],
```
```makefile
INCS_Release := \
        -I/home/.node-gyp/4.1.1/include/node \
        -I/home/.node-gyp/4.1.1/src \
        -I/home/.node-gyp/4.1.1/deps/uv/include \
        -I/home/.node-gyp/4.1.1/deps/v8/include \
        -I$(srcdir)/node_modules/nan \
        -I/usr/local/include \
        -I/usr/local/include/glib-2.0 \
        -I/usr/local/lib/glib-2.0/include \
        -I/usr/local/include/libxml2 \
        -I/usr/local/include/libgsf-1 \
        -I/usr/local/include/GraphicsMagick \
        -I/usr/local/include/orc-0.4 \
        -I/usr/local/include/OpenEXR \
        -I/usr/local/include/pango-1.0 \
        -I/usr/local/include/freetyp2 \
        -I/usr/local/include/harfbuzz \
        -I/usr/local/include/libpng16
```

not sure about a CI you can use in this case but we are planing a production deployment on freebsd, so i'll be keeping an eye on it myself.

**Comment by lovell** [comment]
How odd. My best guess here is that the flavour of `make` on FreeBSD is less forgiving than Linux.

Are you able to submit a PR for the `@` and `sed` change and we'll see what the various Linux-based CI environments "make" of it :)

As an aside, did you install libvips via FreshPorts?

**Comment by lovell** [comment]
Commit dc2b79a allows me to successfully run `npm install lovell/sharp` on FreeBSD.

I used the following commands to install all the dependencies on a fresh 10.2 machine:
```sh
pkg install node www/npm gmake
portsnap fetch
portsnap extract
cd /usr/ports/graphics/vips/
setenv FORCE_PKG_REGISTER
make install clean
```
As an aside, the `cpplint` check run during `npm test` doesn't seem to work on FreeBSD.

**Comment by lovell** [comment]
v0.12.2 now available via npm - thanks for your help @c0decafe
